diff --git a/c10/CMakeLists.txt b/c10/CMakeLists.txt
index 34577ca..dd37686 100644
--- a/c10/CMakeLists.txt
+++ b/c10/CMakeLists.txt
@@ -98,11 +98,8 @@ if(NOT BUILD_LIBTORCHLESS)
   target_link_libraries(c10 PRIVATE nlohmann)
 
   if(C10_USE_NUMA)
-    message(STATUS "NUMA paths:")
-    message(STATUS ${Numa_INCLUDE_DIR})
-    message(STATUS ${Numa_LIBRARIES})
-    target_include_directories(c10 PRIVATE ${Numa_INCLUDE_DIR})
-    target_link_libraries(c10 PRIVATE ${Numa_LIBRARIES})
+    find_package(libnuma REQUIRED)
+    target_link_libraries(c10 PRIVATE libnuma::libnuma)
   else()
     message(STATUS "don't use NUMA")
   endif()
@@ -139,6 +136,11 @@ if(NOT BUILD_LIBTORCHLESS)
       $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/../>
       $<BUILD_INTERFACE:${CMAKE_BINARY_DIR}>
       $<INSTALL_INTERFACE:include>)
+
+  if(USE_SYSTEM_CPUINFO)
+    find_package(cpuinfo REQUIRED)
+    target_link_libraries(c10 PRIVATE cpuinfo::cpuinfo)
+  endif()
 endif()
 
 add_subdirectory(test)
